### PR TITLE
V4H domain reboot

### PR DESCRIPTION
**Pull request by otyshchenko1**
Tested with xl destroy/create and xl reboot, w/ and w/o GPU rendering app/modetest running using custom reduced DomU (with rproc/rvgc and GPU only). No issues were observed. After DomU restart, everything remains working in the context of Shared display feature.

### CODE REVIEW

File: xen/drivers/passthrough/arm/plat-rcar/rproc.c [hunk 1]
```diff
@@ -167,57 +167,62 @@ static void mfis_irq_handler(int irq, void *dev_id, struct cpu_user_regs *regs)
...
     if ( chan >= mfis_data->chan_cnt )
         return -EINVAL;
 
-    mfis_data->domains[chan] = NULL;
+    if ( !vgic_reserve_virq(d, GUEST_MFIS_SPI) )
+        return -EINVAL;
+
+    mfis_data->domains[chan] = d;
+
+    printk("MFIS: Added chan %d for domain %d\n", chan, d->domain_id);
```
Comment (GrygiriiS) [inline xen/drivers/passthrough/arm/plat-rcar/rproc.c]: Log level INFO?

File: xen/drivers/passthrough/arm/plat-rcar/rproc.c [hunk 2]
```diff
@@ -167,57 +167,62 @@ static void mfis_irq_handler(int irq, void *dev_id, struct cpu_user_regs *regs)
...
-                return -EBUSY;
+    vgic_free_virq(d, GUEST_MFIS_SPI);
 
-            writel(1, mfis_data->base + MFIS_IICR(i));
-            return 0;
-        }
+    mfis_data->domains[chan] = NULL;
 
-    return -ENOENT;
+    printk("MFIS: Removed chan %d for domain %d\n", chan, d->domain_id);
```
Comment (GrygiriiS) [inline xen/drivers/passthrough/arm/plat-rcar/rproc.c]: Log level INFO?

**Comment by rshym** [comment]
It seems like a minor misprint in the second commit message: 'otherend.'

File: xen/drivers/passthrough/arm/plat-rcar/rproc.c (hunk 2 above)
Comment (otyshchenko1) [reply xen/drivers/passthrough/arm/plat-rcar/rproc.c]: +1

File: xen/drivers/passthrough/arm/plat-rcar/rproc.c (hunk 1 above)
Comment (otyshchenko1) [reply xen/drivers/passthrough/arm/plat-rcar/rproc.c]: +1

**Pull request merged**